scipy/stats/_stats_py.py

```diff
@@ -3956,7 +3957,109 @@ def __init__(self, msg=None):
         self.args = (msg,)
 
 
-def pearsonr(x, y):
+def _pearsonr_fisher_ci(r, n, confidence_level, alternative):
+    """
+    Compute the confidence interval for Pearson's R.
+
+    Fisher's transformation is used to compute the confidence interval
+    (https://en.wikipedia.org/wiki/Fisher_transformation).
+    """
+    if r == 1:
+        zr = np.inf
+    elif r == -1:
+        zr = -np.inf
+    else:
+        zr = np.arctanh(r)
+    se = np.sqrt(1 / (n - 3))
+
+    if alternative == "two-sided":
+        h = special.ndtri(0.5 + confidence_level/2)
+        zlo = zr - h*se
+        zhi = zr + h*se
+        rlo = np.tanh(zlo)
+        rhi = np.tanh(zhi)
+    elif alternative == "less":
+        h = special.ndtri(confidence_level)
+        zhi = zr + h*se
+        rhi = np.tanh(zhi)
+        rlo = -1.0
+    else:
+        # alternative == "greater":
+        h = special.ndtri(confidence_level)
+        zlo = zr - h*se
+        rlo = np.tanh(zlo)
+        rhi = 1.0
+
+    return ConfidenceInterval(low=rlo, high=rhi)
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+PearsonRResultBase = _make_tuple_bunch('PearsonRResultBase',
+                                       ['statistic', 'pvalue'], [])
+
+
+class PearsonRResult(PearsonRResultBase):
+    """
+    Result of `scipy.stats.pearsonr`
+
+    Attributes
+    ----------
+    statistic : float
+        Pearson product-moment correlation coefficent.
+    pvalue : float
+        P-value.
+    alternative : str
+        The value of `alternative` that was passed to `pearsonr`.
+    n : int
+        The length of `x` and `y` that were given to `pearsonr`.
```

Comment:
Why does this render like:
![image](https://user-images.githubusercontent.com/6570539/153804045-15344c66-e851-44c1-af21-852a2c09e382.png)

I don't want it to link to documentation for attributes `statistic` and `pvalue`:
![image](https://user-images.githubusercontent.com/6570539/153804114-b11f45ac-06f8-4052-b620-1e337a29c140.png)

And I want it to show the description I wrote, not e.g. "Alias for field number 0".

Reply:
@tirthasheshpatel Do you know about this one, too?

Reply:
Sphinx seems to overwrite the docs of property attributes with the one given by `_make_tuple_bunch`. I don't think there is a way to turn the behavior off (at least not just for one specific class). One way is to overwrite the docs given by `_make_tuple_bunch` by adding these lines under the class definition:

```python
PearsonRResult.statistic.__doc__ = ("Pearson product-moment correlation "
                                    "coefficent.")
PearsonRResult.pvalue.__doc__ = "P-value."
```

This will still link to a `pvalue` and `statictic` page since they are properties and not just attributes.

Reply:
Or we can remove the docs from `scipy/_lib/_bunch.py` itself and the code right now should work as you expect i.e. shouldn't link to anything and have the description you have written here. I don't know if docs like `Alias for ...` are useful anyway so this seems to be a better solution.

Reply:
I'll try that. Thank you!

Reply:
W00t! Big thanks @tirthasheshpatel!

Reply:
Looking at your excitement, it seems like this gave you a really hard time! :smiley:

Reply:
Over the years, Sphinx has given me a very, very hard time. Here, I ran into a brick wall, and I expected this PR to stall again as a result. You broke the wall - thank you!